From: Jeremy Bicha <jbicha@debian.org>
Date: Thu, 10 Mar 2022 07:54:26 -0500
Subject: Revert "status/bluetooth: Fix wedged state"

This reverts commit 0531fc54e52407eb11dd04d3e1ab39ef343bdcb8.

Ubuntu 22.04 LTS isn't using the new gnome-bluetooth series
---
 js/ui/status/bluetooth.js | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/js/ui/status/bluetooth.js b/js/ui/status/bluetooth.js
index 2ccca5e..087bc14 100644
--- a/js/ui/status/bluetooth.js
+++ b/js/ui/status/bluetooth.js
@@ -25,7 +25,6 @@ class Indicator extends PanelMenu.SystemIndicator {
         this._indicator = this._addIndicator();
         this._indicator.icon_name = 'bluetooth-active-symbolic';
         this._hadSetupDevices = global.settings.get_boolean(HAD_BLUETOOTH_DEVICES_SETUP);
-        this._client = new GnomeBluetooth.Client();
 
         this._proxy = new RfkillManagerProxy(Gio.DBus.session, BUS_NAME, OBJECT_PATH,
                                              (proxy, error) => {
@@ -43,8 +42,6 @@ class Indicator extends PanelMenu.SystemIndicator {
         this._toggleItem = new PopupMenu.PopupMenuItem('');
         this._toggleItem.connect('activate', () => {
             this._proxy.BluetoothAirplaneMode = !this._proxy.BluetoothAirplaneMode;
-            if (!this._proxy.BluetoothAirplaneMode)
-                this._client.default_adapter_powered = true;
         });
         this._item.menu.addMenuItem(this._toggleItem);
 
@@ -54,6 +51,7 @@ class Indicator extends PanelMenu.SystemIndicator {
         this._syncId = 0;
         this._adapter = null;
 
+        this._client = new GnomeBluetooth.Client();
         this._store = this._client.get_devices();
         this._deviceNotifyConnected = [];
         this._client.connect('device-removed', (c, path) => {
@@ -150,6 +148,6 @@ class Indicator extends PanelMenu.SystemIndicator {
         else
             this._item.label.text = _('Bluetooth On');
 
-        this._toggleItem.label.text = this._client.default_adapter_powered ? _('Turn Off') : _('Turn On');
+        this._toggleItem.label.text = this._proxy.BluetoothAirplaneMode ? _('Turn On') : _('Turn Off');
     }
 });
